sklearn/compose/_column_transformer.py

```diff
@@ -512,7 +512,16 @@ def _hstack(self, Xs):
         Xs : List of numpy arrays, sparse arrays, or DataFrames
         """
         if self.sparse_output_:
-            return sparse.hstack(Xs).tocsr()
+            try:
+                converted_Xs = [check_array(X,
```

Comment:
This looks quite strange and deserves a comment. Basically say that its point is to do dtype casting.